lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -745,16 +746,29 @@ LogicalResult ComponentOp::verify() {
 
   // Verify the component actually does something: has a non-empty Control
   // region, or continuous assignments.
-  bool hasNoControlConstructs =
-      getControlOp().getBodyBlock()->getOperations().empty();
+  bool hasNoControlConstructs = true;
+  getControlOp().walk<WalkOrder::PreOrder>([&](Operation *op) {
+    if (dyn_cast<EnableOp>(op))
+      hasNoControlConstructs = false;
+    if (dyn_cast<InvokeOp>(op))
+      hasNoControlConstructs = false;
+    if (dyn_cast<fsm::MachineOp>(op))
+      hasNoControlConstructs = false;
+    if (hasNoControlConstructs == false)
+      return WalkResult::interrupt();
+    return WalkResult::advance();
```

Comment:
```suggestion
if (isa<EnableOp, InvokeOp, fsm::MachineOp>(op)) {
  hasNoControlConstructs = false;
  return WalkResult::advance();
}
return WalkResult::interrupt();
```

...would be the less verbose way of doing this.

Reply:
Hahaha, graduated and played for a few months, I do remember a cast that could cast multiple types, but I forgot, thanks for the suggestion. It's a bit late now, I'll add what you said tomorrow.